solidity/src/proof_plans/SortMergeJoinExec.presl

```diff
@@ -0,0 +1,418 @@
+    /// * `output_chi_eval` - pointer to the evaluation of a column of 1s with same length as output
+    /// @notice Evaluates a sort merge join execution plan
+    /// @dev Evaluates a sort merge join execution plan
+    /// @param __plan The sort merge join execution plan data
+    /// @param __builder The verification builder
+    /// @return __planOut The remaining plan after processing
+    /// @return __builderOut The verification builder result
+    /// @return __evaluationsPtr The evaluations pointer
+    /// @return __outputLength The length of the output chi evaluation
+    /// @return __outputChiEvaluation The output chi evaluation
+    function __sortMergeJoinEvaluate( // solhint-disable-line gas-calldata-parameters
+    bytes calldata __plan, VerificationBuilder.Builder memory __builder)
+        external
+        pure
+        returns (
+            bytes calldata __planOut,
+            VerificationBuilder.Builder memory __builderOut,
+            uint256[] memory __evaluationsPtr,
+            uint256 __outputLength,
+            uint256 __outputChiEvaluation
+        )
+    {
+        uint256[] memory __evaluations;
+        assembly {
+            // import proof_plan_evaluate from ProofPlan.presl
+            // import err from ../base/Errors.sol
+            // import addmod_bn254, mulmod_bn254, submod_bn254 from ../base/MathUtil.presl
+            // import read_binary from ../base/DataType.presl
+            // import builder_consume_challenge, builder_consume_chi_evaluation_with_length, builder_consume_first_round_mle, builder_get_table_chi_evaluation, builder_produce_zerosum_constraint from ../builder/VerificationBuilder.presl
+            // import proof_expr_evaluate from ../proof_exprs/ProofExpr.presl
+            // import fold_expr_evals from ../proof_gadgets/FoldUtil.presl
+            // import fold_log_star_evaluate_from_column_exprs, fold_log_star_evaluate_from_mles from ../proof_gadgets/FoldLogExpr.presl
+            // import monotonic_verify from ../proof_gadgets/Monotonic.presl
+            // import membership_check_evaluate from ../proof_gadgets/MembershipCheck.presl
+            function evaluate_input_plans(plan_ptr, builder_ptr, evaluations) -> plan_ptr_out, hat_evals, join_evals {
+                // Determine total number of evaluations
+                let num_columns := mload(evaluations)
+                evaluations := add(evaluations, WORD_SIZE)
+
+                // Determine number of columns to join on
+                let num_join_columns := shr(UINT64_PADDING_BITS, calldataload(plan_ptr))
+                plan_ptr := add(plan_ptr, UINT64_SIZE)
+                if sub(num_join_columns, 1) { err(ERR_NUMBER_OF_JOIN_COLUMNS_NOT_ONE) }
+
+                // We need a collection to record which indices are not join indices.
+                // The total number of evaluations from the input should be the number of entries in this collection.
+                let other_indices := mload(FREE_PTR)
```

Comment:
Let n be num_columns (left or right), m be num_join_columns
variable - position from mload(FREE_PTR) over WORD_SIZE - value
other_indices - 0
hat_evals - n - n + 1
target_hat_evals - n + 1
join_evals - 2n + 2 - m
target_join_evals - 2n + 3
FREE_PTR moved to 3n + 3

Slots 1-n: 1 if not join index, 0 if join index